contracts/cw1-subkeys/src/contract.rs

```diff
@@ -130,7 +130,7 @@ where
 pub fn check_staking_permissions(
     staking_msg: &StakingMsg,
     permissions: Permissions,
-) -> Result<bool, ContractError> {
+) -> Result<(), ContractError> {
```

Comment:
Yes, it is exactly why I changed it from bool to ZST - it was hurting my eyes (and also optimizer - in case when one type of Result is ZST, the Result itself becomes transparent).

Reply:
I returned `Result<(), ContractError>` as it captured more failure cases than 1 (a `bool` would just capture one success and one failure case).

We have 1 success case and 4 failure cases. 


Ahhh.... I misread the diff... I thought you made the bool. You removed it. :eyes:
Very good change

Reply:
:+1